File: daily/Sliding Window/3-Longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        seen = set()
        left = 0
        res = 0
        n = len(s)
        for right in range(n):
            while s[right] in seen:
                seen.remove(s[left])
                left += 1
            res = max(res, right - left + 1)
            seen.add(s[right])
        return res

File: daily/Sliding Window/3-Longest-substring-without-repeating-characters.go
package daily

func lengthOfLongestSubstring(s string) int {
	n := len(s)
	left := 0
	res := 0
	seen := make(map[rune]int)
	for right := 0; right < n; right++ {
		seen[rune(s[right])]++
		for seen[rune(s[right])] > 1 {
			seen[rune(s[left])]--
			left++
		}
		res = max(res, right-left+1)
	}
	return res
}
